src/routes/dashboard/index.svelte

```diff
@@ -0,0 +1,417 @@
+<script lang="ts" context="module">
+  import type { Load } from '@sveltejs/kit'
+
+  export const load: Load = ({ props }) => {
+    return {
+      props,
+    }
+  }
+</script>
+
+<script lang="ts">
+  import '@carbon/styles/css/styles.css'
+  import '@carbon/charts/styles.css'
+  import { BarChartStacked, PieChart } from '@carbon/charts-svelte'
+  import {
+    Button,
+    Column,
+    Content,
+    DataTable,
+    DataTableSkeleton,
+    Grid,
+    Row,
+    Tag,
+  } from 'carbon-components-svelte'
+  import {
+    ArrowUp,
+    CaretUp,
+    DocumentDownload,
+    Group,
+  } from 'carbon-icons-svelte'
+  import { toCSV } from './csv/downloadCSV';
+  import { sortChannels } from './helpers/channels'
+  import { filterAnswers, filterQuestions } from './helpers/filter'
+  import { getTopContributors } from './helpers/contributors'
+  import { timeBetweenDates } from './helpers/dates'
+  import FilterMenu from './components/FilterMenu.svelte'
+  import type {
+    Contributor,
+    Contributors,
+    GitHubUser,
+    Questions,
+  } from './types'
+  import ChannelHealth from './components/ChannelHealth.svelte'
+
+  export let channels: string[]
+  export let contributors: Contributors
+  export let gitHubStaff: GitHubUser[]
+  export let memberCount: number
+  export let name: string
+  export let presenceCount: number
+  export let questions: Questions
+
+  let today = new Date()
+  let endDate = today
+  let startDate = new Date(today.getFullYear(), today.getMonth() - 3, 1)
+  let dates: Date[] = timeBetweenDates('months', [startDate, endDate])
+  let filteredQuestions: Map<string, Questions> = filterQuestions(
+    channels,
+    dates,
+    questions
+  )
+  let filteredContributors: Contributor[] = filterAnswers(
+    channels,
+    [dates[0], today],
+    contributors.all
+  )
+  let filteredStaffContributors: Contributor[] = filterAnswers(
+    channels,
+    [dates[0], today],
+    contributors.staff
+  )
+  let topOverallPromise = getTopContributors(contributors.all, gitHubStaff, 9)
+  let topStaffPromise = getTopContributors(contributors.staff, gitHubStaff, 9)
+  const colors = [
```

Comment:
what are these color codes?

Reply:
I got them from the carbon charts defaults by just looking at the color codes used when you don't specify a color. it's so the two pie charts have the same colors for the same channels